arcgis-java-toolkit/src/main/java/com/esri/arcgisruntime/toolkit/UtilityNetworkTraceTool.java

```diff
@@ -0,0 +1,999 @@
+    });
+    completedTraceProperty.set(null);
+    traceResultsProperty.clear();
+  }
+
+  /**
+   * Resets all data relating to the configuration of a new trace. Cancels listenable futures, resets properties
+   * and removes all starting point graphics from the graphics overlay.
+   *
+   * @since 100.15.0
+   */
+  private void resetNewTraceConfigurationProperties() {
+    if (traceInProgressFuture != null) {
+      traceInProgressFuture.cancel(true);
+    }
+    traceInProgressFuture = null;
+    if (fetchFeaturesForElementsFuture != null) {
+      fetchFeaturesForElementsFuture.cancel(true);
+    }
+    fetchFeaturesForElementsFuture = null;
+
+    isTraceInProgressProperty.set(false);
+    isIdentifyInProgressProperty.set(false);
+
+    if (identifyLayersFuture != null) {
+      identifyLayersFuture.cancel(true);
+    }
+    identifyLayersFuture = null;
+    if (queryNamedTraceConfigurationsFuture != null) {
+      queryNamedTraceConfigurationsFuture.cancel(true);
+    }
+    queryNamedTraceConfigurationsFuture = null;
+
+    selectedTraceConfigurationProperty.set(null);
+    traceConfigurationsProperty.set(FXCollections.observableArrayList());
+
+    startingPointsProperty.clear();
+    startingPointsGraphicsOverlay.getGraphics().clear();
+    traceResultInProgress = null;
+    applyStartingPointWarnings();
+  }
+
+  /**
+   * Adds a starting point to be used for a trace from the provided feature.
+   *
+   * <p>
+   * This enables the adding of starting points programmatically, and not just via clicks on the MapView.
+   * For example, feature results from a query, selection, geocode, route, geoprocessing, search, and more.
+   *
+   * <p>
+   * A starting point is only added to the list if a Utility Network has been selected, the feature is part of the
+   * selected utility network and the starting point does not already exist.
+   *
+   * @param feature the feature to use as the basis for the starting point
+   * @throws NullPointerException if feature is null
+   * @since 100.15.0
+   */
+  public void addStartingPoint(ArcGISFeature feature) {
+    addStartingPoint(feature, null);
+  }
+
+  /**
+   * Adds a starting point to be used for a trace from the provided feature and point.
+   *
+   * <p>
+   * This enables the adding of starting points programmatically, and not just via clicks on the MapView.
+   * For example, feature results from a query, selection, geocode, route, geoprocessing, search, and more.
+   *
+   * <p>
+   * A starting point is only added to the list if a Utility Network has been selected, the feature is part of the
+   * selected utility network and the starting point does not already exist.
+   *
+   * @param feature the feature to use as the basis for the starting point
+   * @param mapPoint the point to use - if it's polyline for location along the line
+   * @throws NullPointerException if feature is null
+   * @since 100.15.0
+   */
+  public void addStartingPoint(ArcGISFeature feature, Point mapPoint) {
```

Comment:
From the doc it isn't clear what `mapPoint` is for.

Reply:
The design appears to suggest that when setting starting points programmatically I should be able to set a "fraction along the edge"... fraction suggests to me a numeric value not a point,

Reply:
I've improved the doc and naming here with the existing functionality. I'm going to clarify whether another method is required for providing a fraction along edge value, as opposed to a point, as you raised.

Reply:
Use of a point is what is required, though I've fed your comment back. As per above I've tried to improve the doc here.